hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -497,6 +499,12 @@ public CompletableFuture<Void> terminateJob(long jobId, TerminationMode terminat
                                 + ", should be " + RUNNING);
                     }
 
+                    if ((terminationMode == SUSPEND_GRACEFUL || terminationMode == SUSPEND_FORCEFUL
```

Comment:
this check should be moved to `requestTermination` as there are other places where it called. Some are interesting, eg. `prepareForPassiveClusterState` and may trigger creation of unwanted snapshots. So maybe instead of throwing ISE we might convert unsupported termination mode to a supported one? For analyze job we would convert all suspend/restart/cancel (both graceful and forceful) to cancel_forceful.

Reply:
how about
> maybe instead of throwing ISE we might convert unsupported termination mode to a supported one

IMO that might be simpler that analyzing what will happen if we ignore `requestTermination`